plugins/org.eclipse.n4js.model/model/N4JS.xcore

```diff
@@ -1990,17 +1990,18 @@ class N4TypeVariable extends IdentifiableElement, NamedElement {
 
 	boolean declaredCovariant
 	boolean declaredContravariant
+	boolean declaredOptional
 
 	contains TypeReferenceNode<TypeRef> declaredUpperBoundNode
 
-	/* The default type argument or {@code null} iff this type parameter is non-optional. */
+	/* The default type argument or {@code null}. */
 	contains TypeReferenceNode<TypeRef> defaultArgumentNode
```

Comment:
I'd propose to rename this property to `declaredDefaultArgumentNode`. This would make it easier to remember the subtle difference between this property and `TypeVariable#defaultArgument` in `Types.xcore`: this property can be `null` and should only be interpreted together with `declaredOptional`/`#isOptional()`, whereas the property in `Types.xcore` cannot be `null` and is "self-contained", etc.